classes/amity.py

```diff
@@ -1,45 +1,515 @@
-        pass
+        is_digit = any(char.isdigit() for char in person_name)
+        if is_digit:
+            return "Invalid name. Use letters only"
+        else:
+            person_name = person_name.title()
+            person_type = person_type.lower()
+            if person_type not in ["fellow", "f", "staff", "s"]:
+                return "Invalid role. You can either be a fellow/staff."
+            else:
+                wants_accommodation = (wants_accommodation.lower()
+                                       if wants_accommodation is not None
+                                       else "no")
+                if wants_accommodation not in ["yes", "y", "n", "no"]:
+                    return "Invalid accomodate option. " + \
+                     "It can either be yes or no."
+                else:
+                    return self.save_person(person_name, person_type,
+                                            wants_accommodation)
+
+    def save_person(self, person_name, person_type, wants_accommodation):
+        person_name = person_name.title()
+        if person_type in ["staff", "s"]:
+            if wants_accommodation in ["yes", "y"]:
+                return "Staff cannot be accomodated!"
+            else:
+                new_person = Staff(person_name)
+                new_person.person_id = "SOO" + str(len(self.staffs)+1)
+                self.staffs.append(new_person)
+                self.changes = True
+                print("{} added successfully! Your ID: {}"
+                      .format(person_name, new_person.person_id))
+                return self.allocate_office(new_person)
+
+        else:
+            new_person = Fellow(person_name, wants_accommodation)
+            new_person.person_id = "FOO" + str(len(self.fellows)+1)
+            self.fellows.append(new_person)
+            self.changes = True
+            print("{} added successfully! Your ID: {}"
+                  .format(person_name, new_person.person_id))
+            if wants_accommodation in ["no", "n"]:
+                return self.allocate_office(new_person)
+            else:
+                print(self.allocate_office(new_person))
+                return self.allocate_living_space(new_person)
+
+    def allocate_office(self, new_person):
+        """A method that allocates an office"""
+        office_with_space = []
+        for office in self.offices:
+            if len(office.occupants) < office.room_capacity:
+                office_with_space.append(office)
+        if office_with_space:
+            random_office = choice(office_with_space)
+            random_office.occupants.append(new_person.person_name)
+            new_person.office = random_office
+            return "Allocated office: {}\n".format(random_office.room_name)
+        else:
+            self.waiting_list["office"].append(new_person)
+            self.waiting_list["office"] = list(
+             set(self.waiting_list["office"]))
```

Comment:
line 117 is similar to line 135 despite one being for office waiting list and the other for living spaces.